cleanlab_studio/internal/util.py

```diff
@@ -258,11 +258,25 @@ def tracked_func(*args: Any, **kwargs: Any) -> Any:
                     if track_all_frames:
                         cleanlab_traceback = trace_str
                     else:
+                        # remove stack frames for user code
                         cleanlab_match = re.search("File.*cleanlab", trace_str)
                         cleanlab_traceback = (
                             trace_str[cleanlab_match.start() :] if cleanlab_match else ""
                         )
 
+                        def replace_file_newline(match: re.Match[str]) -> str:
+                            if 'cleanlab-studio' not in match.group(0):
+                                return "File\n"
+                            else:
+                                return match.group(0)
+                        # First filter out "File" followed by any characters except newline until a newline, without cleanlab-studio
+                        pattern1 = re.compile(r"File.*?\n")
+                        cleanlab_traceback = pattern1.sub(replace_file_newline, cleanlab_traceback)
+
+                        # Then filter out anything between "File" followed by any characters (no newline) until 'cleanlab-studio'
+                        pattern2 = re.compile(r"File([^\n]*?)cleanlab-studio")
+                        cleanlab_traceback = pattern2.sub(r"File cleanlab-studio", cleanlab_traceback)
```

Comment:
nit: should use `r"File \"cleanlab-studio"` since the filepaths in stack traces are surrounded by quotes